### PR TITLE
Fix issue that loads properties from configs that start with the same

### PR DESCRIPTION
**Pull request by baumgarb**
name like the config that is being processed at the moment

Resolves: #22

### CODE REVIEW

File: pkg/config/config_test.go
Comment (didiladi) [inline pkg/config/config_test.go:69]: Can you also add a test for two properties with groups in their names?

E.g.:
```
m["dashboard"] = make(map[string]string)
m["dashboard"]["myProperty"] = "test"
m["dashboard.devEnv"]  = make(map[string]string)
m["dashboard.devEnv"]["myProperty"] = "test for dev"

m["dashboard-availability.devEnv"] = make(map[string]string)
```

**Review by didiladi — CHANGES_REQUESTED** [review]
Cool, thanks for your contribution. The change looks good. Would be cool to have an additional test here, thanks!

File: pkg/config/config_test.go
Comment (baumgarb) [reply pkg/config/config_test.go:69]: Thanks, yeah that makes a ton of sense. I've created the test and updated the PR already.

File: pkg/config/config_test.go
Comment (didiladi) [inline pkg/config/config_test.go:104]: Sorry about nitpicking, but these two assertions would make sense here:

```
assert.Equal(t, "B", properties["dashboard.dev"]["prop1"])
assert.Equal(t, "C", properties["dashboard.dev"]["prop2"])
```

Comment (baumgarb) [reply pkg/config/config_test.go:104]: I thought about that but I deliberately chose against it. I can add them if really want, but I would rather create a separate test for this because if I add these assertions we're testing different things with one tiny unit test. The test name says it's only testing for exact match of config name, but the assertions you'd like to add here are testing a different thing. They're testing if more specific properties, if any, are overwriting more general ones. 

Or what do you think? I'll wait for your answer before adding / modifying the tests here.

Comment (didiladi) [reply pkg/config/config_test.go:104]: I see. Yes, you are completely right. By doing that, your test would be not tesing just a single thing anymore. Would be cool to have such a test with the assertions, since we are missing it currently.

Comment (baumgarb) [reply pkg/config/config_test.go:105]: All right, I've implemented two more tests. Can you have a look? Thanks!

Comment (didiladi) [reply pkg/config/config_test.go:105]: Cool, thank you!

**Comment by didiladi** [comment]
> ###  Integration tests failed on ubuntu-latest

The failing integration tests have nothing to do with the changes in this PR.

**Review by didiladi — APPROVED** [review]

**Comment by baumgarb** [comment]
Do I need to do anything to merge this PR now? I think it says "Only those with write access to this repository can merge pull requests."

**Comment by didiladi** [comment]
> Do I need to do anything to merge this PR now? I think it says "Only those with write access to this repository can merge pull requests."

Hey, I'm currently in the progress of fixing the integration tests (#63). Once this is done, I'll re-run the integration tests for your change and merge the PR :)

**Pull request merged**
